### PR TITLE
feat: avm/res/cache/redis - Add access policies

### PR DESCRIPTION
**Pull request by jlinn-microsoft**
## Description
Adds access policies and access policy assignments to the Redis module.

## Pipeline Reference

| Pipeline |
| -------- |
|    [avm.res.cache.redis](https://github.com/jlinn-microsoft/bicep-registry-modules/actions/workflows/avm.res.cache.redis.yml)      |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by jlinn-microsoft** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: avm/res/cache/redis/tests/e2e/entra-id/main.test.bicep [hunk 1]
```diff
@@ -31,6 +31,15 @@ resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
   location: resourceLocation
 }
 
+module dependencies 'dependencies.bicep' = {
```
Comment (hundredacres) [inline avm/res/cache/redis/tests/e2e/entra-id/main.test.bicep:34]: Update name of module

`module nestedDependencies ''dependencies.bicep' = {`

File: avm/res/cache/redis/tests/e2e/entra-id/main.test.bicep [hunk 2]
```diff
@@ -31,6 +31,15 @@ resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
...
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-dependencies'
```
Comment (hundredacres) [inline avm/res/cache/redis/tests/e2e/entra-id/main.test.bicep:36]: Update to

`name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'`

File: avm/res/cache/redis/tests/e2e/entra-id/main.test.bicep [hunk 3]
```diff
@@ -46,6 +55,22 @@ module testDeployment '../../../main.bicep' = [
       redisConfiguration: {
         'aad-enabled': 'true'
       }
+      accessPolicyAssignments: [
+        {
+          accessPolicyName: 'Data Contributor'
+          objectId: dependencies.outputs.managedIdentityPrincipalId
```
Comment (hundredacres) [inline avm/res/cache/redis/tests/e2e/entra-id/main.test.bicep:61]: Update to

```
objectId: nestedDependencies.outputs.managedIdentityPrincipalId
objectIdAlias: nestedDependencies.outputs.managedIdentityName
```

**Review by hundredacres — CHANGES_REQUESTED** [review]

File: avm/res/cache/redis/tests/e2e/entra-id/main.test.bicep (hunk 3 above)
Comment (jlinn-microsoft) [reply avm/res/cache/redis/tests/e2e/entra-id/main.test.bicep]: Done, done, and done.

**Review by hundredacres — APPROVED** [review]

**Pull request merged**